src/utils/ContextSettingsUser.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState, createContext, useContext } from 'react';
+import { FC } from 'react';
+import { API, headers } from 'API/api';
+import { GlobalContext } from './GlobalContext';
+
+interface ContextSettingsUserProviderProps {
+  children: React.ReactNode;
+}
+interface ContextSettingsUserValue {
+  defaultWordLanguage: string;
+  setDefaultWordLanguage: React.Dispatch<React.SetStateAction<string>>;
+  defaultWordLanguageTranslate: string;
+  setDefaultWordLanguageTranslate: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const UserSettingsContext = createContext<ContextSettingsUserValue>(
+  {} as ContextSettingsUserValue
+);
+
+const UserSettingsProvider: FC<ContextSettingsUserProviderProps> = ({ children }) => {
+  const { isLoginUser } = useContext(GlobalContext);
+  const [defaultWordLanguage, setDefaultWordLanguage] = useState('English');
+  const [defaultWordLanguageTranslate, setDefaultWordLanguageTranslate] = useState('Polish');
+  
```

Comment:
te wartości powinny przyjść z bazy.
Na dole masz funkcje pobierającą settings. i to z niej powinnaś pobrać te dane.
I z tego co pamiętam default Language jest dla nas nie ważny. My potrzebujemy znać tylko Translate language, czyli ten, którego user chce się nauczyć. Nie ma dla nas znaczenia z jakiego startuje.